### PR TITLE
feat(slider): add support for custom label formatting

### PR DESCRIPTION
**Pull request by jcfranco**
**Related Issue:** #4004

## Summary

Adds `labelFormatter` prop to allow customization of handle and tick labels.

```ts
interface CalciteSlider {

  /** 
    * Formatter for custom handle/ticks labels.
    * 
    * If the returned value is a string, it will get used in the label, 
    * and if it is `undefined`, it will use the default, formatted label.
    */
  labelFormatter(

    // the associated label value
    value: number, 

    // the label type
    type: "min" | "max" | "value" | "tick" 

    // the default formatter (configured with corresponding locale, numbering system and group separator
    defaultFormatter: (value: number): string;

  ): string | undefined;
}
```

### CODE REVIEW

File: packages/calcite-components/src/components/slider/slider.stories.ts [hunk 1]
```diff
@@ -462,3 +462,64 @@ export const fillPlacements = (): string => html`
...
+    step="1"
+    min-label="Temperature"
+  ></calcite-slider>
+
+  <script>
+    const singleValueSlider = document.getElementById("singleFormattedLabelSlider");
+
+    singleValueSlider.labelFormatter = function (value, type) {
+      if (type === "value") {
+        return value < 60 ? "🥶" : value > 80 ? "🥵" : "😎";
```
Comment (driskull) [inline packages/calcite-components/src/components/slider/slider.stories.ts:499]: 🆒

File: packages/calcite-components/src/components/slider/slider.stories.ts [hunk 2]
```diff
@@ -462,3 +462,64 @@ export const fillPlacements = (): string => html`
...
+
+      if (type === "tick") {
+        return value === singleValueSlider.min ? "Cold" : value === singleValueSlider.max ? "Hot" : undefined;
+      }
+    };
+
+    const minMaxValueSlider = document.getElementById("minMaxFormattedLabelSlider");
+
+    minMaxValueSlider.labelFormatter = function (value, type) {
+      console.log("yo");
```
Comment (driskull) [inline packages/calcite-components/src/components/slider/slider.stories.ts:510]: remove console.log yo

**Review by driskull — APPROVED** [review]
😎 🌞 👍

File: packages/calcite-components/src/components/slider/slider.stories.ts (hunk 2 above)
Comment (jcfranco) [reply packages/calcite-components/src/components/slider/slider.stories.ts:510]: Thanks for the catch! 🎣

**Pull request merged**